uwblueprint/sistering
# Add status field to Posting and update Posting types and resolvers

## Ticket link

Closes #93



## Implementation description
* Added `status` column in the `Posting` table as enum of values:
  * DRAFT
  * PUBLISHED
  * ARCHIVED
* Generated new migration file for status field
* Updated `PostingRequestDTO` and `PostingResponseDTO` and corresponding GraphQL types with the status field
* Modified `PostingService` CRUD methods to read/write the status value



## Steps to test
1. We need row(s) in the skills, branches, and employees tables for testing which can be created using the GraphQL mutations or with raw SQL:
```bash
docker exec -it sistering_db /bin/bash
psql -U postgres -d sistering
INSERT INTO branches (name) VALUES ('Branch1');
INSERT INTO skills (name) VALUES ('Skill1');
INSERT INTO employees ("userId", branch_id) VALUES ('1','1');
```
2. Run `docker exec -it sistering_backend /bin/bash -c "npx prisma generate"` 
3. Go to http://localhost:5000/graphql and verify that we can read/write the status field using GraphQL queries and mutations



## What should reviewers focus on?
* Verify that we can read/write status field as expected


## Checklist
